tests/catch/unit/device/hipSetGetDevice.cc

```diff
@@ -83,6 +87,46 @@ TEST_CASE("Unit_hipGetSetDevice_MultiThreaded") {
   HIP_CHECK_THREAD_FINALIZE();
 }
 
+TEST_CASE("Unit_hipSetGetDevice_Positive_Threaded_Basic") {
+  class HipSetGetDeviceThreadedTest : public ThreadedZigZagTest<HipSetGetDeviceThreadedTest> {
+   public:
+    void TestPart1() { HIP_CHECK(hipSetDevice(0)); }
+
+    void TestPart2() {
+      HIP_CHECK_THREAD(hipSetDevice(1));
+      HIP_CHECK_THREAD(hipMalloc(&ptr, 2 * 1024 * 1024));
```

Comment:
hipFree the allocated memory in ptr

Reply:
The suggested change is implemented. Please proceed with the review.